osu.Game/Rulesets/Objects/Drawables/DrawableHitObject.cs

```diff
@@ -134,6 +137,68 @@ protected override void LoadComplete()
             updateState(ArmedState.Idle, true);
         }
 
+        private void apply(HitObject hitObject)
+        {
+#pragma warning disable 618 // can be removed 20200417
+            if (GetType().GetMethod(nameof(AddNested), BindingFlags.NonPublic | BindingFlags.Instance)?.DeclaringType != typeof(DrawableHitObject))
+                return;
+#pragma warning restore 618
+
+            if (nestedHitObjects.IsValueCreated)
+            {
+                nestedHitObjects.Value.Clear();
+                ClearNestedHitObjects();
+            }
+
+            foreach (var h in hitObject.NestedHitObjects)
+            {
+                var drawableNested = CreateNestedHitObject(h) ?? throw new InvalidOperationException($"{nameof(CreateNestedHitObject)} returned null for {h.GetType().ReadableName()}.");
```

Comment:
are you sure we don't want to allow null here? potentially some hitobjects don't need drawable pairs. for example slider end circles (which currently have a drawable but don't necessarily need it; it shows nothing).

Reply:
discussed IRL, makes sense (DHOs are required to handle judgement results).